src/app/posts/post-detail/post-detail.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Post } from 'src/app/models/post.model';
+import { PostsService } from 'src/app/services/posts.service';
+
+@Component({
+  selector: 'app-post-detail',
+  templateUrl: './post-detail.component.html',
+  styleUrls: ['./post-detail.component.css']
+})
+export class PostDetailComponent implements OnInit {
+  id: number
+  posts: Post[] = [];
+  comments: string[] = [];
+  i:number;
+  constructor(private route: ActivatedRoute, private postService: PostsService) { }
+
+  ngOnInit(): void {
+    this.route.params.subscribe(
+      (params: Params) => {
+        this.id = +params['id'];
+       
+      }
+    )
+    this.postService.getPost(this.id).subscribe(
+      (post: Post) => {
+        this.posts.push(post);
+        console.log(post);
+      }
+    )
+
+    this.postService.getPostsComments(this.id).subscribe(
```

Comment:
can we create a model PostWithComments that includes Post and Comment[], and create getPostWithComments method that merges calls to getPost and getComments?